app/controllers/course/announcements_controller.rb

```diff
@@ -4,6 +4,10 @@ class Course::AnnouncementsController < Course::ModuleController
   def index #:nodoc:
     @announcements = @announcements.includes(:creator).sorted_by_sticky.sorted_by_date
     @announcements = @announcements.page(params[:page])
+    # mark each announcement in this page as read
+    @announcements.each do |ann|
```

Comment:
https://github.com/ledermann/unread/blob/e4b8ae9a39aa2cac25fd4b2382d4a8242d4910fc/lib/unread/readable.rb

you're looking for `Model.mark_array_as_read!(items)` it can be made more efficient, but that should do for now